lua/autorun/client/cl_chip_lister.lua

```diff
@@ -0,0 +1,321 @@
+local TOGGLE_DIST_SQR = TOGGLE_DIST ^ 2
+
+local matChipLister = CreateMaterial( MATERIAL_NAME, "UnlitGeneric", {
+    ["$basetexture"] = RENDER_TARGET_NAME,
+    ["$model"] = 1,
+} )
+
+local isValid = IsValid
+local rawset = rawset
+local rawget = rawget
+local _colorToHSV = ColorToHSV
+local _hsvToColor = HSVToColor
+local stringLen = string.len
+local stringSub = string.sub
+local stringFormat = string.format
+local teamGetColor = team.GetColor
+local getClass
+local getTeam
+
+local FONT_DATA = {
+    font = "Roboto Mono",
+    extended = false,
+    size = FONT_SIZE,
+    weight = 500,
+    blursize = 0,
+    scanlines = 0,
+    antialias = true,
+    underline = false,
+    italic = false,
+    strikeout = false,
+    symbol = false,
+    rotary = false,
+    shadow = false,
+    Additive = false,
+    outline = false,
+}
+
+do
+    local entityMeta = FindMetaTable( "Entity" )
+
+    getClass = entityMeta.GetClass
+
+
+    local playerMeta = FindMetaTable( "Player" )
+
+    getTeam = playerMeta.Team
+
+
+    if not file.Exists( "resource/fonts/RobotoMono.ttf", "MOD" ) then
+        local files, folders = file.Find( "resource/fonts/*", "THIRDPARTY" )
+        local robotoExists = false
+
+        for _, v in ipairs( files ) do
+            if v == "RobotoMono.ttf" then
+                robotoExists = true
+
+                break
+            end
+        end
+
+        if not robotoExists then
+            FONT_DATA.font = "Arial"
+        end
+    end
+
+    surface.CreateFont( FONT_NAME, FONT_DATA )
+
+    surface.SetFont( FONT_NAME )
+    INFO_OFFSET_OWNER = -surface.GetTextSize( STR_TOTAL .. stringFormat( CPUS_FORMAT, 0 ) .. STR_MICROSECONDS )
+    INFO_OFFSET_CHIP = -surface.GetTextSize( table.GetKeys( CHIP_COLORS )[1] .. " " .. stringFormat( CPUS_FORMAT, 0 ) .. STR_MICROSECONDS )
+end
+
+local LISTER_ENABLED = CreateClientConVar( "cfc_chiplister_enabled", 1, true, false, "Enables the Expression2/Starfall chip lister." )
+
+local listerEnabled = LISTER_ENABLED:GetBool()
+
+
+include( "cfc_chip_lister/client/cl_hud.lua" )
+
+
+local function colorToHSV( color )
+    return Color( _colorToHSV( color ) )
+end
+
+local function hsvToColor( color )
+    return _hsvToColor( color.r, color.g, color.b )
+end
+
+local function formatCPUs( num )
+    local usageStr = stringFormat( CPUS_FORMAT, num or 0 )
+    local leadStr = ""
+    local numStr = ""
+    local leadCount = 0
+
+    for i = 1, stringLen( usageStr ) do
+        if stringSub( usageStr, i, i ) == "0" then
+            leadStr = leadStr .. "0"
+            leadCount = leadCount + 1
+        else
+            return leadStr, stringSub( usageStr, leadCount + 1 )
+        end
+    end
+
+    return leadStr, ""
+end
+
+local function getTeamColor( ply )
+    if ply == ID_WOLRD or not isValid( ply ) then
+        return COLOR_WORLD
+    end
+
+    return teamGetColor( getTeam( ply ) )
+end
+
+local function fadeColor( color, fadeOverride )
+    local hsv = colorToHSV( color )
```

Comment:
Why put it into a Color just to reference the r g b components of the color?
Also youre using color which is meant to store Red Green Blue Alpha values. To store HSV values.

Reply:
could you not just do `h, s, v = ColorToHSV( color )`

Reply:
It's to package things into a single object instead of having to manually switch between formats and track each value separately. It's a generalized method, but since it's only being used for one thing, I'll trim it down.